misc/python/materialize/mzcompose.py

```diff
@@ -1214,6 +1214,38 @@ def __init__(
         )
 
 
+class Kgen(PythonService):
+    def __init__(
+        self,
+        name: str = "kgen",
+        mzbuild: str = "kgen",
+        depends_on: List[str] = ["kafka"],
+    ) -> None:
+        entrypoint = [
+            "kgen",
+            "--bootstrap-server=kafka:9092",
+        ]
+
+        if "schema-registry" in depends_on:
+            entrypoint.append("--schema-registry-url=http://schema-registry:8081")
+
+        super().__init__(
+            name=name,
+            config={
+                "mzbuild": mzbuild,
+                "depends_on": depends_on,
+                "entrypoint": [
+                    "bash",
+                    "-O",
+                    "extglob",
+                    "-c",
+                    " ".join(entrypoint),
+                    "bash",
+                ],
```

Comment:
This is weird testdrive crap that you should ditch! Just the entrypoint as you've defined it on L1224 is great.